File: LeetCode/C/66-array-加一.cpp
/*
 * @lc app=leetcode.cn id=66 lang=cpp
 *
 * [66] 加一
 */

// @lc code=start
class Solution {
  public:
    vector<int> plusOne(vector<int> &digits) {
        int carry = 1;
        int len = digits.size();
        for (int i = len - 1; i >= 0; --i) {
            int t = carry + digits[i];
            digits[i] = t % 10;
            carry = t / 10;
        }
        if (carry != 0)
            digits.insert(digits.begin(), carry);
        return digits;
    }
};
// @lc code=end

File: LeetCode/Go/66-array-加一.go
/*
 * @lc app=leetcode.cn id=66 lang=golang
 *
 * [66] 加一
 */
package leetcode

// @lc code=start
func plusOne(digits []int) []int {
	var carry = 1
	for i := len(digits) - 1; i >= 0; i-- {
		carry += digits[i]
		digits[i] = carry % 10
		carry /= 10
	}
	if carry != 0 {
		digits = append([]int{carry}, digits...)
	}
	return digits
}

// @lc code=end
